pkgs/applications/office/trilium/default.nix

```diff
@@ -101,6 +102,9 @@ in {
       exec ./node/bin/node src/www
       EOF
       chmod a+x $out/bin/trilium-server
+
+      cd $out/share/trilium-server
+      patch -p1 < "${ ./0001-Use-console-logger-instead-of-rolling-files.patch }"
```

Comment:
You can use the patchPhase built into stdenv to apply the patch. Note, the patch will apply before the installPhase:

```
diff --git a/pkgs/applications/office/trilium/default.nix b/pkgs/applications/office/trilium/default.nix
index 579fe1a7c58..2c8c2b47eab 100644
--- a/pkgs/applications/office/trilium/default.nix
+++ b/pkgs/applications/office/trilium/default.nix
@@ -91,6 +91,8 @@ in {
       libxkbfile
     ];
 
+    patches = [ ./0001-Use-console-logger-instead-of-rolling-files.patch ];
+
     installPhase = ''
       mkdir -p $out/bin
       mkdir -p $out/share/trilium-server
@@ -104,7 +106,6 @@ in {
       chmod a+x $out/bin/trilium-server
 
       cd $out/share/trilium-server
-      patch -p1 < "${ ./0001-Use-console-logger-instead-of-rolling-files.patch }"
     '';
   };
 }
```

Reply:
Done